File: leetcode/38.py
class Solution(object):
     def countAndSay(self, n):
        s = '1'
        for _ in range(n-1):
            let, temp, count = s[0], '', 0
            for l in s:
                if let == l:
                    count += 1
                else:
                    temp += str(count)+let
                    let = l
                    count = 1
            temp += str(count)+let
            s = temp
        return s

File: leetcode/38.cpp
class Solution {
public:
    string countAndSay(int n) {
        string ans = "1";
        for (int i = 1; i < n; ++i) {
            string now = "";
            for (int j = 0; j < ans.size(); ++j) {
                int cnt = 1;
                while (j + 1 < ans.size() && ans[j + 1] == ans[j]) {
                    ++j, ++cnt;
                }
                now += to_string(cnt) + ans[j];
            }
            ans = now;
        }
        return ans;
    }
};